gradio/cli/commands/components/publish.py

```diff
@@ -217,125 +184,81 @@ def _publish(
-            else Path(".") / "pyproject.toml"
-        )
+        pyproject_toml_path = source_dir / "pyproject.toml"
 
         try:
             pyproject_toml = parse(pyproject_toml_path.read_text())
             package_name = pyproject_toml["project"]["name"]  # type: ignore
         except Exception:
-            (package_name, version) = wheel_file.name.split("-")[:2]
-
-        try:
-            latest_release = httpx.get(
-                f"https://pypi.org/pypi/{package_name}/json"
-            ).json()["info"]["version"]
-        except Exception:
-            latest_release = None
+            (package_name, _) = wheel_file.name.split("-")[:2]
 
         if not demo_dir:
             raise ValueError("demo_dir must be set")
-        demo_path = resolve_demo(demo_dir)
-
-        if prefer_local or not latest_release:
-            additional_reqs = [wheel_file.name]
-        else:
-            additional_reqs = [f"{package_name}=={latest_release}"]
-        if (demo_dir / "requirements.txt").exists():
-            reqs = (demo_dir / "requirements.txt").read_text().splitlines()
-            reqs += additional_reqs
-        else:
-            reqs = additional_reqs
 
-        color_from, color_to = random.choice(colors), random.choice(colors)
-        package_name, version = wheel_file.name.split("-")[:2]
+        package_name, _ = wheel_file.name.split("-")[:2]
         with tempfile.TemporaryDirectory() as tempdir:
             shutil.copytree(
                 str(demo_dir),
                 str(tempdir),
                 dirs_exist_ok=True,
             )
-            if source_dir:
+            if upload_source:
                 shutil.copytree(
                     str(source_dir),
                     str(Path(tempdir) / "src"),
                     dirs_exist_ok=True,
                     ignore=_ignore,
                 )
-            reqs_txt = Path(tempdir) / "requirements.txt"
-            reqs_txt.write_text("\n".join(reqs))
-            readme = Path(tempdir) / "README.md"
-            template = ""
-            if upload_source and source_dir:
-                pyproject_toml = parse((source_dir / "pyproject.toml").read_text())
-                keywords = pyproject_toml["project"].get("keywords", [])  # type: ignore
-                keywords = [
-                    k
-                    for k in keywords
-                    if k not in {"gradio-custom-component", "gradio custom component"}
-                ]
-                if keywords:
-                    template = "," + ",".join(keywords)
-                name = pyproject_toml["project"]["name"]  # type: ignore
-                description = pyproject_toml["project"]["description"]  # type: ignore
 
-            readme_text = README_CONTENTS.format(
-                package_name=package_name,
-                version=version,
-                color_from=color_from,
-                color_to=color_to,
-                template=template,
+            shutil.copyfile(
```

Comment:
(not related to this line-change)

Inside `with tempfile.TemporaryDirectory() as tempdir:` context, it would be good to create a `.gitignore` file with common rules. Otherwise the uploaded Space contains a bunch of useless files (`__pycache__`, `.ruff_cache`, `.mypy_cache`, maybe some front-end build?, etc.).

FYI, `huggingface_hub`'s `upload_folder` respects the gitignore file at the root of the directory but not all gitignores in subdirectories. This is an `hfh`'s limitation I'm aware of but not planned to be fixed (yet?).  That's why you would need to add more rules to the root gitignore files to avoid folders like `.ruff_cache`.

Reply:
We create a gitignore file in the `create` step. I will add some more common cases like `mypy_cache` and `ruff_cache` and also copy this file to the `tempdir` used here.